src/main/java/org/folio/service/InvoiceStorageService.java

```diff
@@ -17,55 +16,48 @@
 
 import org.apache.commons.lang3.StringUtils;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
 import org.folio.dao.invoice.InvoiceDAO;
 import org.folio.rest.jaxrs.model.Document;
 import org.folio.rest.jaxrs.model.DocumentCollection;
 import org.folio.rest.jaxrs.model.Invoice;
+import org.folio.rest.jaxrs.model.InvoiceAuditEvent;
 import org.folio.rest.jaxrs.model.InvoiceDocument;
 import org.folio.rest.jaxrs.resource.InvoiceStorage.GetInvoiceStorageInvoicesDocumentsByIdResponse;
-import org.folio.rest.jaxrs.resource.InvoiceStorage.PutInvoiceStorageInvoicesByIdResponse;
 import org.folio.rest.persist.DBClient;
 import org.folio.rest.persist.PgUtil;
+import org.folio.service.audit.AuditOutboxService;
 
 import io.vertx.core.AsyncResult;
 import io.vertx.core.Context;
 import io.vertx.core.Handler;
 import io.vertx.ext.web.handler.HttpException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
 
+@Log4j2
+@RequiredArgsConstructor
 public class InvoiceStorageService {
 
-  private static final Logger log = LogManager.getLogger(InvoiceStorageService.class);
   private static final String INVOICE_ID_MISMATCH_ERROR_MESSAGE = "Invoice id mismatch";
 
   private final InvoiceDAO invoiceDAO;
-
-  public InvoiceStorageService(InvoiceDAO invoiceDAO) {
-    this.invoiceDAO = invoiceDAO;
-  }
+  private final AuditOutboxService auditOutboxService;
 
   public void postInvoiceStorageInvoices(Invoice invoice, Handler<AsyncResult<Response>> asyncResultHandler,
       Context vertxContext, Map<String, String> headers) {
     try {
       vertxContext.runOnContext(v -> {
         log.info("postInvoiceStorageInvoices:: Creating a new invoice by id: {}", invoice.getId());
-
-        DBClient client = new DBClient(vertxContext, headers);
-        client.startTx()
-          .compose(t -> invoiceDAO.createInvoice(invoice, client))
-          .compose(t -> client.endTx())
-          .onComplete(reply -> {
-            if (reply.failed()) {
-              // The result of rollback operation is not so important, main failure cause is used to build the response
-              client.rollbackTransaction().onComplete(res -> asyncResultHandler.handle(buildErrorResponse(
-                reply.cause())));
-            } else {
-              log.info("postInvoiceStorageInvoices:: Preparing response to client");
-              asyncResultHandler.handle(
-                buildResponseWithLocation(headers.get(OKAPI_URL), INVOICE_PREFIX + invoice.getId(), invoice)
-              );
-            }
+        new DBClient(vertxContext, headers).getPgClient()
+          .withTrans(conn -> invoiceDAO.createInvoice(invoice, conn)
+            .compose(invoiceId -> auditOutboxService.saveInvoiceOutboxLog(conn, invoice, InvoiceAuditEvent.Action.CREATE, headers)))
+          .onSuccess(s -> {
+            log.info("postInvoiceStorageInvoices:: Successfully created a new invoice by id: {}", invoice.getId());
```

Comment:
the same about invoking auditOutboxService.processOutboxEventLogs(okapiHeaders);
Basically it should be invoked in all places where saveOutboxLog() is invoked